File: data/src/main/java/com/ykb/app/cryptotrader/data/dao/TradeBotDao.java
package com.ykb.app.cryptotrader.data.dao;

import com.ykb.app.cryptotrader.data.model.TradeBot;
import com.ykb.app.cryptotrader.data.repo.TradeBotRepo;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public final class TradeBotDao extends Dao<TradeBot, Long> {

    public TradeBotDao(TradeBotRepo repo) {
        super(repo);
    }

    public List<TradeBot> findAllNonTerminated() {
        return ((TradeBotRepo) repo).findAllNonTerminated();
    }

}

File: library/data/data-domain/src/main/kotlin/com/ykb/app/cryptotrader/data/domain/dao/TradeBotDao.kt
package com.ykb.app.cryptotrader.data.domain.dao

import com.ykb.app.cryptotrader.data.base.Dao
import com.ykb.app.cryptotrader.data.domain.model.TradeBot
import com.ykb.app.cryptotrader.data.domain.repo.TradeBotRepo
import com.ykb.app.cryptotrader.utils.enums.TradeBotStatus
import org.springframework.stereotype.Component

@Component
class TradeBotDao(repo: TradeBotRepo) : Dao<TradeBot, Int>(repo) {

    fun findByStatus(status: TradeBotStatus): List<TradeBot> {
        return (repo as TradeBotRepo).findByStatus(status)
    }

}